### ISSUE TITLE
Make the preview text of new messages bold

### ISSUE THREAD

**Issue by th0rgall** [WelcometoMyGarden/welcometomygarden#276]
See the suggestion here https://github.com/WelcometoMyGarden/welcometomygarden/issues/41#issuecomment-657235857.

This does not happen yet!

**Comment by carlos-ds** [comment]
I'll try to recap here because perhaps I'm missing some information.

For this issue I should:

- When a user sends a new chat message, add or update an item in the `chats` collection (as is) but also add a new boolean field to track if the last message has already been read by the receiver e.g. `lastMessageRead` (or `lastMessageUnread`?)
- When a user opens his chat page, check the value for `lastMessageRead` of all conversations and display the preview text as bold if `false`
- When the receiver opens a message for which `lastMessageRead` is `false`, update the value of `lastMessageRead` for that conversation to `true` and retrieve the latest data from the `chats` collection again

Does that sound right?

**Comment by th0rgall** [comment]
Thanks for looking into this! It seems I was also missing information.

I had only skimmed the chat code in the past, and I always assumed that the "NEW" label on a new chat meant that I had not read a chat yet. That's why I thought it would be easy to also make the chat title bold, if the label is shown.

But that label only reflects that you're  _composing a new chat_ (one that does not have a message yet), it's unrelated to unread messages:
https://github.com/WelcometoMyGarden/welcometomygarden/blob/ccc05889c9d040810678cfdec61b3bdf6bcff3c4/src/lib/components/Chat/ConversationCard.svelte#L12-L16

As you figured, we indeed don't store anything yet related to the read status of the receiver.

Your implementation plan sounds like it would work! The only concern I have is that we would store the "read status" in the shared chat document. If the sender is a hacker inspecting their client-side JS code, they could determine whether the receiver has read their last message or not. It's not the end of the world, but it's something to be aware of.

An alternative is storing the read status in the `users-private` document. We already have a `unreads` subcollection there, which is only used for chat notification emails (to make sure that we don't send many emails when someone send a burst of multiple messages). The documents therein could probably be adjusted to also support read statuses, in a more private way. Probably this takes a bit more work too.

I wouldn't call this an ~~Good~~ easy first issue anymore, since it became a full-stack problem, but if you're willing to take a shot at either of these two implementation strategies, I'd be happy to review that code!

Note that I started fixing #271 in https://github.com/WelcometoMyGarden/welcometomygarden/tree/fix/chat-streaming, it's probably best to branch off from there.

**Comment by th0rgall** [comment]
I thought it was better not to over-engineer this, and I've now implemented the simpler route, thanks for the proposal @carlos-ds!

In addition to a `lastMessageUnread` field in the `chats` collections (I made it `lastMessageSeen`), I also had to add a `lastMessageSender` field with the sender UID. This is for performance reasons: we can now still display the whole chat list without fetching messages of every chat. Only messages of the opened chat need to be fetched. This is presumably the same reason why we had a `lastMessage` field to begin with. 

One other current limitation to be aware of (in addition to what I pointed out before) is that a single `lastMessageSeen` field can not be used for an "Mark as unread" feature, because this solution assumes that only the receiver of the last message is interested in a read/unread (seen/not seen) status. In contrast, a "Mark as unread" action should also be able to be performed by the last message sender. In that case we probably want to give all chat members their own read/not read status with a migration.
